{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b492339c-7a3c-4e55-a515-f9a86e20a44c",
   "metadata": {},
   "source": [
    "# Gastric Warriors Differential gene expression analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c611cf59-304e-4336-900c-819ac8bdac4d",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "Importing correct files: Using bash\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fa6fbc16-09cb-41b2-baef-2a8cddee4204",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working directory set to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/Data/\n",
      "RNAseq-data in /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p Data/\n",
    "workdir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/Data/\"\n",
    "echo \"Working directory set to $workdir\"\n",
    "\n",
    "RNAlist=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt\"\n",
    "RNAdata=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/\"\n",
    "echo \"RNAseq-data in ${RNAdata}\"\n",
    "\n",
    "\n",
    "grep \"Met\" $RNAlist | cut -f 1 > patients.txt\n",
    "grep \"Hp\" $RNAlist | cut -f 1 >> patients.txt\n",
    "\n",
    "find $RNAdata -type f -name \"*_1.fastq.gz\" | grep -Ff patients.txt | xargs -I {} cp {} \"$workdir\"\n",
    "find $RNAdata -type f -name \"*_2.fastq.gz\" | grep -Ff patients.txt | xargs -I {} cp {} \"$workdir\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "467126d3-f738-4a95-8e3f-3f47ce105660",
   "metadata": {},
   "source": [
    "Create a CSV with patient list and health condition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6e7258f-ad5c-4336-9754-299be544c7d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Create patient list\n",
    "RNAlist=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt\"\n",
    "\n",
    "# Define the output file\n",
    "OUTPUT=\"patient_list.csv\"\n",
    "# Add CSV header\n",
    "echo \"PatientNumber,HealthCondition\" > $OUTPUT\n",
    "\n",
    "# Create patient_list.txt with \"Met\" and \"Hp\" conditions\n",
    "awk '{print $1, \",Met\"}' <(grep \"Met\" $RNAlist | cut -f1) >> $OUTPUT\n",
    "awk '{print $1, \",Hp\"}' <(grep \"Hp\" $RNAlist | cut -f1) >> $OUTPUT\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccf9c36f-dbff-414f-a8c6-d2218d7ed4ac",
   "metadata": {},
   "source": [
    "Now decompress all the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a5acecd9-5780-4b4f-85b8-2326f9eedf39",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#set the working directory\n",
    "workdir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/Data/\"\n",
    "\n",
    "#use gzip to decrompress the files\n",
    "gzip -d \"${workdir}\"*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0c8c86e-9873-410c-b93e-97e4d3f4ebd2",
   "metadata": {},
   "source": [
    "# MultiQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0b15720e-c086-4482-8aaf-1232d5f21787",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "null\n",
      "null\n",
      "Started analysis of ERR950168_1.fastq\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "Approx 5% complete for ERR950168_1.fastq\n",
      "Approx 10% complete for ERR950168_1.fastq\n",
      "Approx 15% complete for ERR950168_1.fastq\n",
      "Approx 20% complete for ERR950168_1.fastq\n",
      "Approx 25% complete for ERR950168_1.fastq\n",
      "Approx 30% complete for ERR950168_1.fastq\n",
      "Approx 35% complete for ERR950168_1.fastq\n",
      "Approx 40% complete for ERR950168_1.fastq\n",
      "Approx 45% complete for ERR950168_1.fastq\n",
      "Approx 50% complete for ERR950168_1.fastq\n",
      "Approx 55% complete for ERR950168_1.fastq\n",
      "Approx 60% complete for ERR950168_1.fastq\n",
      "Approx 65% complete for ERR950168_1.fastq\n",
      "Approx 70% complete for ERR950168_1.fastq\n",
      "Approx 75% complete for ERR950168_1.fastq\n",
      "Approx 80% complete for ERR950168_1.fastq\n",
      "Approx 85% complete for ERR950168_1.fastq\n",
      "Approx 90% complete for ERR950168_1.fastq\n",
      "Approx 95% complete for ERR950168_1.fastq\n",
      "Analysis complete for ERR950168_1.fastq\n",
      "Started analysis of ERR950168_2.fastq\n",
      "Approx 5% complete for ERR950168_2.fastq\n",
      "Approx 10% complete for ERR950168_2.fastq\n",
      "Approx 15% complete for ERR950168_2.fastq\n",
      "Approx 20% complete for ERR950168_2.fastq\n",
      "Approx 25% complete for ERR950168_2.fastq\n",
      "Approx 30% complete for ERR950168_2.fastq\n",
      "Approx 35% complete for ERR950168_2.fastq\n",
      "Approx 40% complete for ERR950168_2.fastq\n",
      "Approx 45% complete for ERR950168_2.fastq\n",
      "Approx 50% complete for ERR950168_2.fastq\n",
      "Approx 55% complete for ERR950168_2.fastq\n",
      "Approx 60% complete for ERR950168_2.fastq\n",
      "Approx 65% complete for ERR950168_2.fastq\n",
      "Approx 70% complete for ERR950168_2.fastq\n",
      "Approx 75% complete for ERR950168_2.fastq\n",
      "Approx 80% complete for ERR950168_2.fastq\n",
      "Approx 85% complete for ERR950168_2.fastq\n",
      "Approx 90% complete for ERR950168_2.fastq\n",
      "Approx 95% complete for ERR950168_2.fastq\n",
      "Analysis complete for ERR950168_2.fastq\n",
      "Started analysis of ERR950169_1.fastq\n",
      "Approx 5% complete for ERR950169_1.fastq\n",
      "Approx 10% complete for ERR950169_1.fastq\n",
      "Approx 15% complete for ERR950169_1.fastq\n",
      "Approx 20% complete for ERR950169_1.fastq\n",
      "Approx 25% complete for ERR950169_1.fastq\n",
      "Approx 30% complete for ERR950169_1.fastq\n",
      "Approx 35% complete for ERR950169_1.fastq\n",
      "Approx 40% complete for ERR950169_1.fastq\n",
      "Approx 45% complete for ERR950169_1.fastq\n",
      "Approx 50% complete for ERR950169_1.fastq\n",
      "Approx 55% complete for ERR950169_1.fastq\n",
      "Approx 60% complete for ERR950169_1.fastq\n",
      "Approx 65% complete for ERR950169_1.fastq\n",
      "Approx 70% complete for ERR950169_1.fastq\n",
      "Approx 75% complete for ERR950169_1.fastq\n",
      "Approx 80% complete for ERR950169_1.fastq\n",
      "Approx 85% complete for ERR950169_1.fastq\n",
      "Approx 90% complete for ERR950169_1.fastq\n",
      "Approx 95% complete for ERR950169_1.fastq\n",
      "Analysis complete for ERR950169_1.fastq\n",
      "Started analysis of ERR950169_2.fastq\n",
      "Approx 5% complete for ERR950169_2.fastq\n",
      "Approx 10% complete for ERR950169_2.fastq\n",
      "Approx 15% complete for ERR950169_2.fastq\n",
      "Approx 20% complete for ERR950169_2.fastq\n",
      "Approx 25% complete for ERR950169_2.fastq\n",
      "Approx 30% complete for ERR950169_2.fastq\n",
      "Approx 35% complete for ERR950169_2.fastq\n",
      "Approx 40% complete for ERR950169_2.fastq\n",
      "Approx 45% complete for ERR950169_2.fastq\n",
      "Approx 50% complete for ERR950169_2.fastq\n",
      "Approx 55% complete for ERR950169_2.fastq\n",
      "Approx 60% complete for ERR950169_2.fastq\n",
      "Approx 65% complete for ERR950169_2.fastq\n",
      "Approx 70% complete for ERR950169_2.fastq\n",
      "Approx 75% complete for ERR950169_2.fastq\n",
      "Approx 80% complete for ERR950169_2.fastq\n",
      "Approx 85% complete for ERR950169_2.fastq\n",
      "Approx 90% complete for ERR950169_2.fastq\n",
      "Approx 95% complete for ERR950169_2.fastq\n",
      "Analysis complete for ERR950169_2.fastq\n",
      "Started analysis of ERR950170_1.fastq\n",
      "Approx 5% complete for ERR950170_1.fastq\n",
      "Approx 10% complete for ERR950170_1.fastq\n",
      "Approx 15% complete for ERR950170_1.fastq\n",
      "Approx 20% complete for ERR950170_1.fastq\n",
      "Approx 25% complete for ERR950170_1.fastq\n",
      "Approx 30% complete for ERR950170_1.fastq\n",
      "Approx 35% complete for ERR950170_1.fastq\n",
      "Approx 40% complete for ERR950170_1.fastq\n",
      "Approx 45% complete for ERR950170_1.fastq\n",
      "Approx 50% complete for ERR950170_1.fastq\n",
      "Approx 55% complete for ERR950170_1.fastq\n",
      "Approx 60% complete for ERR950170_1.fastq\n",
      "Approx 65% complete for ERR950170_1.fastq\n",
      "Approx 70% complete for ERR950170_1.fastq\n",
      "Approx 75% complete for ERR950170_1.fastq\n",
      "Approx 80% complete for ERR950170_1.fastq\n",
      "Approx 85% complete for ERR950170_1.fastq\n",
      "Approx 90% complete for ERR950170_1.fastq\n",
      "Approx 95% complete for ERR950170_1.fastq\n",
      "Analysis complete for ERR950170_1.fastq\n",
      "Started analysis of ERR950170_2.fastq\n",
      "Approx 5% complete for ERR950170_2.fastq\n",
      "Approx 10% complete for ERR950170_2.fastq\n",
      "Approx 15% complete for ERR950170_2.fastq\n",
      "Approx 20% complete for ERR950170_2.fastq\n",
      "Approx 25% complete for ERR950170_2.fastq\n",
      "Approx 30% complete for ERR950170_2.fastq\n",
      "Approx 35% complete for ERR950170_2.fastq\n",
      "Approx 40% complete for ERR950170_2.fastq\n",
      "Approx 45% complete for ERR950170_2.fastq\n",
      "Approx 50% complete for ERR950170_2.fastq\n",
      "Approx 55% complete for ERR950170_2.fastq\n",
      "Approx 60% complete for ERR950170_2.fastq\n",
      "Approx 65% complete for ERR950170_2.fastq\n",
      "Approx 70% complete for ERR950170_2.fastq\n",
      "Approx 75% complete for ERR950170_2.fastq\n",
      "Approx 80% complete for ERR950170_2.fastq\n",
      "Approx 85% complete for ERR950170_2.fastq\n",
      "Approx 90% complete for ERR950170_2.fastq\n",
      "Approx 95% complete for ERR950170_2.fastq\n",
      "Analysis complete for ERR950170_2.fastq\n",
      "Started analysis of ERR950177_1.fastq\n",
      "Approx 5% complete for ERR950177_1.fastq\n",
      "Approx 10% complete for ERR950177_1.fastq\n",
      "Approx 15% complete for ERR950177_1.fastq\n",
      "Approx 20% complete for ERR950177_1.fastq\n",
      "Approx 25% complete for ERR950177_1.fastq\n",
      "Approx 30% complete for ERR950177_1.fastq\n",
      "Approx 35% complete for ERR950177_1.fastq\n",
      "Approx 40% complete for ERR950177_1.fastq\n",
      "Approx 45% complete for ERR950177_1.fastq\n",
      "Approx 50% complete for ERR950177_1.fastq\n",
      "Approx 55% complete for ERR950177_1.fastq\n",
      "Approx 60% complete for ERR950177_1.fastq\n",
      "Approx 65% complete for ERR950177_1.fastq\n",
      "Approx 70% complete for ERR950177_1.fastq\n",
      "Approx 75% complete for ERR950177_1.fastq\n",
      "Approx 80% complete for ERR950177_1.fastq\n",
      "Approx 85% complete for ERR950177_1.fastq\n",
      "Approx 90% complete for ERR950177_1.fastq\n",
      "Approx 95% complete for ERR950177_1.fastq\n",
      "Analysis complete for ERR950177_1.fastq\n",
      "Started analysis of ERR950177_2.fastq\n",
      "Approx 5% complete for ERR950177_2.fastq\n",
      "Approx 10% complete for ERR950177_2.fastq\n",
      "Approx 15% complete for ERR950177_2.fastq\n",
      "Approx 20% complete for ERR950177_2.fastq\n",
      "Approx 25% complete for ERR950177_2.fastq\n",
      "Approx 30% complete for ERR950177_2.fastq\n",
      "Approx 35% complete for ERR950177_2.fastq\n",
      "Approx 40% complete for ERR950177_2.fastq\n",
      "Approx 45% complete for ERR950177_2.fastq\n",
      "Approx 50% complete for ERR950177_2.fastq\n",
      "Approx 55% complete for ERR950177_2.fastq\n",
      "Approx 60% complete for ERR950177_2.fastq\n",
      "Approx 65% complete for ERR950177_2.fastq\n",
      "Approx 70% complete for ERR950177_2.fastq\n",
      "Approx 75% complete for ERR950177_2.fastq\n",
      "Approx 80% complete for ERR950177_2.fastq\n",
      "Approx 85% complete for ERR950177_2.fastq\n",
      "Approx 90% complete for ERR950177_2.fastq\n",
      "Approx 95% complete for ERR950177_2.fastq\n",
      "Analysis complete for ERR950177_2.fastq\n",
      "Started analysis of ERR950178_1.fastq\n",
      "Approx 5% complete for ERR950178_1.fastq\n",
      "Approx 10% complete for ERR950178_1.fastq\n",
      "Approx 15% complete for ERR950178_1.fastq\n",
      "Approx 20% complete for ERR950178_1.fastq\n",
      "Approx 25% complete for ERR950178_1.fastq\n",
      "Approx 30% complete for ERR950178_1.fastq\n",
      "Approx 35% complete for ERR950178_1.fastq\n",
      "Approx 40% complete for ERR950178_1.fastq\n",
      "Approx 45% complete for ERR950178_1.fastq\n",
      "Approx 50% complete for ERR950178_1.fastq\n",
      "Approx 55% complete for ERR950178_1.fastq\n",
      "Approx 60% complete for ERR950178_1.fastq\n",
      "Approx 65% complete for ERR950178_1.fastq\n",
      "Approx 70% complete for ERR950178_1.fastq\n",
      "Approx 75% complete for ERR950178_1.fastq\n",
      "Approx 80% complete for ERR950178_1.fastq\n",
      "Approx 85% complete for ERR950178_1.fastq\n",
      "Approx 90% complete for ERR950178_1.fastq\n",
      "Approx 95% complete for ERR950178_1.fastq\n",
      "Analysis complete for ERR950178_1.fastq\n",
      "Started analysis of ERR950178_2.fastq\n",
      "Approx 5% complete for ERR950178_2.fastq\n",
      "Approx 10% complete for ERR950178_2.fastq\n",
      "Approx 15% complete for ERR950178_2.fastq\n",
      "Approx 20% complete for ERR950178_2.fastq\n",
      "Approx 25% complete for ERR950178_2.fastq\n",
      "Approx 30% complete for ERR950178_2.fastq\n",
      "Approx 35% complete for ERR950178_2.fastq\n",
      "Approx 40% complete for ERR950178_2.fastq\n",
      "Approx 45% complete for ERR950178_2.fastq\n",
      "Approx 50% complete for ERR950178_2.fastq\n",
      "Approx 55% complete for ERR950178_2.fastq\n",
      "Approx 60% complete for ERR950178_2.fastq\n",
      "Approx 65% complete for ERR950178_2.fastq\n",
      "Approx 70% complete for ERR950178_2.fastq\n",
      "Approx 75% complete for ERR950178_2.fastq\n",
      "Approx 80% complete for ERR950178_2.fastq\n",
      "Approx 85% complete for ERR950178_2.fastq\n",
      "Approx 90% complete for ERR950178_2.fastq\n",
      "Approx 95% complete for ERR950178_2.fastq\n",
      "Analysis complete for ERR950178_2.fastq\n",
      "Started analysis of ERR950179_1.fastq\n",
      "Approx 5% complete for ERR950179_1.fastq\n",
      "Approx 10% complete for ERR950179_1.fastq\n",
      "Approx 15% complete for ERR950179_1.fastq\n",
      "Approx 20% complete for ERR950179_1.fastq\n",
      "Approx 25% complete for ERR950179_1.fastq\n",
      "Approx 30% complete for ERR950179_1.fastq\n",
      "Approx 35% complete for ERR950179_1.fastq\n",
      "Approx 40% complete for ERR950179_1.fastq\n",
      "Approx 45% complete for ERR950179_1.fastq\n",
      "Approx 50% complete for ERR950179_1.fastq\n",
      "Approx 55% complete for ERR950179_1.fastq\n",
      "Approx 60% complete for ERR950179_1.fastq\n",
      "Approx 65% complete for ERR950179_1.fastq\n",
      "Approx 70% complete for ERR950179_1.fastq\n",
      "Approx 75% complete for ERR950179_1.fastq\n",
      "Approx 80% complete for ERR950179_1.fastq\n",
      "Approx 85% complete for ERR950179_1.fastq\n",
      "Approx 90% complete for ERR950179_1.fastq\n",
      "Approx 95% complete for ERR950179_1.fastq\n",
      "Analysis complete for ERR950179_1.fastq\n",
      "Started analysis of ERR950179_2.fastq\n",
      "Approx 5% complete for ERR950179_2.fastq\n",
      "Approx 10% complete for ERR950179_2.fastq\n",
      "Approx 15% complete for ERR950179_2.fastq\n",
      "Approx 20% complete for ERR950179_2.fastq\n",
      "Approx 25% complete for ERR950179_2.fastq\n",
      "Approx 30% complete for ERR950179_2.fastq\n",
      "Approx 35% complete for ERR950179_2.fastq\n",
      "Approx 40% complete for ERR950179_2.fastq\n",
      "Approx 45% complete for ERR950179_2.fastq\n",
      "Approx 50% complete for ERR950179_2.fastq\n",
      "Approx 55% complete for ERR950179_2.fastq\n",
      "Approx 60% complete for ERR950179_2.fastq\n",
      "Approx 65% complete for ERR950179_2.fastq\n",
      "Approx 70% complete for ERR950179_2.fastq\n",
      "Approx 75% complete for ERR950179_2.fastq\n",
      "Approx 80% complete for ERR950179_2.fastq\n",
      "Approx 85% complete for ERR950179_2.fastq\n",
      "Approx 90% complete for ERR950179_2.fastq\n",
      "Approx 95% complete for ERR950179_2.fastq\n",
      "Analysis complete for ERR950179_2.fastq\n",
      "Started analysis of ERR950185_1.fastq\n",
      "Approx 5% complete for ERR950185_1.fastq\n",
      "Approx 10% complete for ERR950185_1.fastq\n",
      "Approx 15% complete for ERR950185_1.fastq\n",
      "Approx 20% complete for ERR950185_1.fastq\n",
      "Approx 25% complete for ERR950185_1.fastq\n",
      "Approx 30% complete for ERR950185_1.fastq\n",
      "Approx 35% complete for ERR950185_1.fastq\n",
      "Approx 40% complete for ERR950185_1.fastq\n",
      "Approx 45% complete for ERR950185_1.fastq\n",
      "Approx 50% complete for ERR950185_1.fastq\n",
      "Approx 55% complete for ERR950185_1.fastq\n",
      "Approx 60% complete for ERR950185_1.fastq\n",
      "Approx 65% complete for ERR950185_1.fastq\n",
      "Approx 70% complete for ERR950185_1.fastq\n",
      "Approx 75% complete for ERR950185_1.fastq\n",
      "Approx 80% complete for ERR950185_1.fastq\n",
      "Approx 85% complete for ERR950185_1.fastq\n",
      "Approx 90% complete for ERR950185_1.fastq\n",
      "Approx 95% complete for ERR950185_1.fastq\n",
      "Analysis complete for ERR950185_1.fastq\n",
      "Started analysis of ERR950185_2.fastq\n",
      "Approx 5% complete for ERR950185_2.fastq\n",
      "Approx 10% complete for ERR950185_2.fastq\n",
      "Approx 15% complete for ERR950185_2.fastq\n",
      "Approx 20% complete for ERR950185_2.fastq\n",
      "Approx 25% complete for ERR950185_2.fastq\n",
      "Approx 30% complete for ERR950185_2.fastq\n",
      "Approx 35% complete for ERR950185_2.fastq\n",
      "Approx 40% complete for ERR950185_2.fastq\n",
      "Approx 45% complete for ERR950185_2.fastq\n",
      "Approx 50% complete for ERR950185_2.fastq\n",
      "Approx 55% complete for ERR950185_2.fastq\n",
      "Approx 60% complete for ERR950185_2.fastq\n",
      "Approx 65% complete for ERR950185_2.fastq\n",
      "Approx 70% complete for ERR950185_2.fastq\n",
      "Approx 75% complete for ERR950185_2.fastq\n",
      "Approx 80% complete for ERR950185_2.fastq\n",
      "Approx 85% complete for ERR950185_2.fastq\n",
      "Approx 90% complete for ERR950185_2.fastq\n",
      "Approx 95% complete for ERR950185_2.fastq\n",
      "Analysis complete for ERR950185_2.fastq\n",
      "Started analysis of ERR950186_1.fastq\n",
      "Approx 5% complete for ERR950186_1.fastq\n",
      "Approx 10% complete for ERR950186_1.fastq\n",
      "Approx 15% complete for ERR950186_1.fastq\n",
      "Approx 20% complete for ERR950186_1.fastq\n",
      "Approx 25% complete for ERR950186_1.fastq\n",
      "Approx 30% complete for ERR950186_1.fastq\n",
      "Approx 35% complete for ERR950186_1.fastq\n",
      "Approx 40% complete for ERR950186_1.fastq\n",
      "Approx 45% complete for ERR950186_1.fastq\n",
      "Approx 50% complete for ERR950186_1.fastq\n",
      "Approx 55% complete for ERR950186_1.fastq\n",
      "Approx 60% complete for ERR950186_1.fastq\n",
      "Approx 65% complete for ERR950186_1.fastq\n",
      "Approx 70% complete for ERR950186_1.fastq\n",
      "Approx 75% complete for ERR950186_1.fastq\n",
      "Approx 80% complete for ERR950186_1.fastq\n",
      "Approx 85% complete for ERR950186_1.fastq\n",
      "Approx 90% complete for ERR950186_1.fastq\n",
      "Approx 95% complete for ERR950186_1.fastq\n",
      "Analysis complete for ERR950186_1.fastq\n",
      "Started analysis of ERR950186_2.fastq\n",
      "Approx 5% complete for ERR950186_2.fastq\n",
      "Approx 10% complete for ERR950186_2.fastq\n",
      "Approx 15% complete for ERR950186_2.fastq\n",
      "Approx 20% complete for ERR950186_2.fastq\n",
      "Approx 25% complete for ERR950186_2.fastq\n",
      "Approx 30% complete for ERR950186_2.fastq\n",
      "Approx 35% complete for ERR950186_2.fastq\n",
      "Approx 40% complete for ERR950186_2.fastq\n",
      "Approx 45% complete for ERR950186_2.fastq\n",
      "Approx 50% complete for ERR950186_2.fastq\n",
      "Approx 55% complete for ERR950186_2.fastq\n",
      "Approx 60% complete for ERR950186_2.fastq\n",
      "Approx 65% complete for ERR950186_2.fastq\n",
      "Approx 70% complete for ERR950186_2.fastq\n",
      "Approx 75% complete for ERR950186_2.fastq\n",
      "Approx 80% complete for ERR950186_2.fastq\n",
      "Approx 85% complete for ERR950186_2.fastq\n",
      "Approx 90% complete for ERR950186_2.fastq\n",
      "Approx 95% complete for ERR950186_2.fastq\n",
      "Analysis complete for ERR950186_2.fastq\n",
      "Started analysis of ERR950187_1.fastq\n",
      "Approx 5% complete for ERR950187_1.fastq\n",
      "Approx 10% complete for ERR950187_1.fastq\n",
      "Approx 15% complete for ERR950187_1.fastq\n",
      "Approx 20% complete for ERR950187_1.fastq\n",
      "Approx 25% complete for ERR950187_1.fastq\n",
      "Approx 30% complete for ERR950187_1.fastq\n",
      "Approx 35% complete for ERR950187_1.fastq\n",
      "Approx 40% complete for ERR950187_1.fastq\n",
      "Approx 45% complete for ERR950187_1.fastq\n",
      "Approx 50% complete for ERR950187_1.fastq\n",
      "Approx 55% complete for ERR950187_1.fastq\n",
      "Approx 60% complete for ERR950187_1.fastq\n",
      "Approx 65% complete for ERR950187_1.fastq\n",
      "Approx 70% complete for ERR950187_1.fastq\n",
      "Approx 75% complete for ERR950187_1.fastq\n",
      "Approx 80% complete for ERR950187_1.fastq\n",
      "Approx 85% complete for ERR950187_1.fastq\n",
      "Approx 90% complete for ERR950187_1.fastq\n",
      "Approx 95% complete for ERR950187_1.fastq\n",
      "Approx 100% complete for ERR950187_1.fastq\n",
      "Analysis complete for ERR950187_1.fastq\n",
      "Started analysis of ERR950187_2.fastq\n",
      "Approx 5% complete for ERR950187_2.fastq\n",
      "Approx 10% complete for ERR950187_2.fastq\n",
      "Approx 15% complete for ERR950187_2.fastq\n",
      "Approx 20% complete for ERR950187_2.fastq\n",
      "Approx 25% complete for ERR950187_2.fastq\n",
      "Approx 30% complete for ERR950187_2.fastq\n",
      "Approx 35% complete for ERR950187_2.fastq\n",
      "Approx 40% complete for ERR950187_2.fastq\n",
      "Approx 45% complete for ERR950187_2.fastq\n",
      "Approx 50% complete for ERR950187_2.fastq\n",
      "Approx 55% complete for ERR950187_2.fastq\n",
      "Approx 60% complete for ERR950187_2.fastq\n",
      "Approx 65% complete for ERR950187_2.fastq\n",
      "Approx 70% complete for ERR950187_2.fastq\n",
      "Approx 75% complete for ERR950187_2.fastq\n",
      "Approx 80% complete for ERR950187_2.fastq\n",
      "Approx 85% complete for ERR950187_2.fastq\n",
      "Approx 90% complete for ERR950187_2.fastq\n",
      "Approx 95% complete for ERR950187_2.fastq\n",
      "Approx 100% complete for ERR950187_2.fastq\n",
      "Analysis complete for ERR950187_2.fastq\n"
     ]
    }
   ],
   "source": [
    "#First generate fastQC reports, all files in Data and output to /subset_fastqc/\n",
    "! mkdir -p before_trim_fastqc\n",
    "! fastqc Data/*.fastq -o before_trim_fastqc/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7230731f-8a91-4d92-83f3-9c8c23a2753d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=566247;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ ðŸ” \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/before_trim_fastqc\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m36/36\u001b[0m  stqc/ERR950187_2_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 18 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : multiqc_before_trim_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : multiqc_before_trim.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "#Take entire subset_fastqc/ and generate multiqc-file\n",
    "! multiqc before_trim_fastqc/ -n multiqc_before_trim.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfc7d074-0afe-4621-889d-7d4866590f0e",
   "metadata": {},
   "source": [
    "# Preprocessing using FastP\n",
    "> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b1e71cd2-4aaa-4712-ba8b-f17a4f039dca",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking: Data/ERR950168_1.fastq and Data/ERR950168_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 24913290\n",
      "total bases: 2516242290\n",
      "Q20 bases: 2354746125(93.5819%)\n",
      "Q30 bases: 2141558358(85.1094%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 24913290\n",
      "total bases: 2516242290\n",
      "Q20 bases: 2355137804(93.5974%)\n",
      "Q30 bases: 2196623898(87.2978%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 19493030\n",
      "total bases: 1538088835\n",
      "Q20 bases: 1536952633(99.9261%)\n",
      "Q30 bases: 1511522404(98.2728%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 19493030\n",
      "total bases: 1596778475\n",
      "Q20 bases: 1595365240(99.9115%)\n",
      "Q30 bases: 1574038254(98.5759%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 38986060\n",
      "reads failed due to low quality: 44620\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 10795900\n",
      "reads with adapter trimmed: 5938208\n",
      "bases trimmed due to adapters: 119420244\n",
      "\n",
      "Duplication rate: 29.7874%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 101\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950168_1.fastq -I Data/ERR950168_2.fastq -o trimmed_fastq/ERR950168_1_trimmed_fastq.gz -O trimmed_fastq/ERR950168_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 55 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950168\n",
      "Checking: Data/ERR950169_1.fastq and Data/ERR950169_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 33285817\n",
      "total bases: 3361867517\n",
      "Q20 bases: 3026738306(90.0315%)\n",
      "Q30 bases: 2691689990(80.0653%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 33285817\n",
      "total bases: 3361867517\n",
      "Q20 bases: 3084045147(91.7361%)\n",
      "Q30 bases: 2844590197(84.6134%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 20859542\n",
      "total bases: 1563426927\n",
      "Q20 bases: 1562271896(99.9261%)\n",
      "Q30 bases: 1535629009(98.222%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 20859542\n",
      "total bases: 1630666733\n",
      "Q20 bases: 1628899215(99.8916%)\n",
      "Q30 bases: 1604600902(98.4015%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 41719084\n",
      "reads failed due to low quality: 131232\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 24721318\n",
      "reads with adapter trimmed: 14139482\n",
      "bases trimmed due to adapters: 456404715\n",
      "\n",
      "Duplication rate: 30.6051%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 1\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950169_1.fastq -I Data/ERR950169_2.fastq -o trimmed_fastq/ERR950169_1_trimmed_fastq.gz -O trimmed_fastq/ERR950169_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 69 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950169\n",
      "Checking: Data/ERR950170_1.fastq and Data/ERR950170_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 29270698\n",
      "total bases: 2956340498\n",
      "Q20 bases: 2669177147(90.2865%)\n",
      "Q30 bases: 2364754032(79.9892%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 29270698\n",
      "total bases: 2956340498\n",
      "Q20 bases: 2668102608(90.2502%)\n",
      "Q30 bases: 2425930397(82.0586%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 17459211\n",
      "total bases: 1327111801\n",
      "Q20 bases: 1326064156(99.9211%)\n",
      "Q30 bases: 1302057051(98.1121%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 17459211\n",
      "total bases: 1360604093\n",
      "Q20 bases: 1358986258(99.8811%)\n",
      "Q30 bases: 1336439402(98.224%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 34918422\n",
      "reads failed due to low quality: 144542\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 23478432\n",
      "reads with adapter trimmed: 9430582\n",
      "bases trimmed due to adapters: 317744657\n",
      "\n",
      "Duplication rate: 22.8644%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 1\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950170_1.fastq -I Data/ERR950170_2.fastq -o trimmed_fastq/ERR950170_1_trimmed_fastq.gz -O trimmed_fastq/ERR950170_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 60 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950170\n",
      "Checking: Data/ERR950177_1.fastq and Data/ERR950177_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 23796570\n",
      "total bases: 2403453570\n",
      "Q20 bases: 2230341208(92.7973%)\n",
      "Q30 bases: 2025357185(84.2686%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 23796570\n",
      "total bases: 2403453570\n",
      "Q20 bases: 2168563767(90.227%)\n",
      "Q30 bases: 1984382392(82.5638%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 18275536\n",
      "total bases: 1452264503\n",
      "Q20 bases: 1451073313(99.918%)\n",
      "Q30 bases: 1427744923(98.3116%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 18275536\n",
      "total bases: 1458017169\n",
      "Q20 bases: 1456543912(99.899%)\n",
      "Q30 bases: 1434566244(98.3916%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 36551072\n",
      "reads failed due to low quality: 66466\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 10975602\n",
      "reads with adapter trimmed: 4117300\n",
      "bases trimmed due to adapters: 85524357\n",
      "\n",
      "Duplication rate: 36.3873%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 1\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950177_1.fastq -I Data/ERR950177_2.fastq -o trimmed_fastq/ERR950177_1_trimmed_fastq.gz -O trimmed_fastq/ERR950177_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 53 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950177\n",
      "Checking: Data/ERR950178_1.fastq and Data/ERR950178_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 26360829\n",
      "total bases: 2662443729\n",
      "Q20 bases: 2491504647(93.5796%)\n",
      "Q30 bases: 2276198546(85.4928%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 26360829\n",
      "total bases: 2662443729\n",
      "Q20 bases: 2465041612(92.5857%)\n",
      "Q30 bases: 2282141132(85.716%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 20881202\n",
      "total bases: 1640115402\n",
      "Q20 bases: 1638807067(99.9202%)\n",
      "Q30 bases: 1613661571(98.3871%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 20881202\n",
      "total bases: 1664943747\n",
      "Q20 bases: 1663400776(99.9073%)\n",
      "Q30 bases: 1641067760(98.566%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 41762404\n",
      "reads failed due to low quality: 67206\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 10892048\n",
      "reads with adapter trimmed: 9310678\n",
      "bases trimmed due to adapters: 198733667\n",
      "\n",
      "Duplication rate: 16.5969%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 1\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950178_1.fastq -I Data/ERR950178_2.fastq -o trimmed_fastq/ERR950178_1_trimmed_fastq.gz -O trimmed_fastq/ERR950178_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 58 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950178\n",
      "Checking: Data/ERR950179_1.fastq and Data/ERR950179_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 17671067\n",
      "total bases: 1784777767\n",
      "Q20 bases: 1676134627(93.9128%)\n",
      "Q30 bases: 1536986985(86.1164%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 17671067\n",
      "total bases: 1784777767\n",
      "Q20 bases: 1671672802(93.6628%)\n",
      "Q30 bases: 1569923924(87.9619%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 14090115\n",
      "total bases: 1130789760\n",
      "Q20 bases: 1129909765(99.9222%)\n",
      "Q30 bases: 1111912428(98.3306%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 14090115\n",
      "total bases: 1179797755\n",
      "Q20 bases: 1178728902(99.9094%)\n",
      "Q30 bases: 1163681086(98.6339%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 28180230\n",
      "reads failed due to low quality: 28828\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 7133076\n",
      "reads with adapter trimmed: 3956364\n",
      "bases trimmed due to adapters: 76677944\n",
      "\n",
      "Duplication rate: 24.0002%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 101\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950179_1.fastq -I Data/ERR950179_2.fastq -o trimmed_fastq/ERR950179_1_trimmed_fastq.gz -O trimmed_fastq/ERR950179_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 39 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950179\n",
      "Checking: Data/ERR950185_1.fastq and Data/ERR950185_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 23865721\n",
      "total bases: 2410437821\n",
      "Q20 bases: 2257720947(93.6644%)\n",
      "Q30 bases: 2070944522(85.9157%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 23865721\n",
      "total bases: 2410437821\n",
      "Q20 bases: 2228514716(92.4527%)\n",
      "Q30 bases: 2070720003(85.9064%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 19404480\n",
      "total bases: 1543599854\n",
      "Q20 bases: 1542358470(99.9196%)\n",
      "Q30 bases: 1519356189(98.4294%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 19404480\n",
      "total bases: 1574224299\n",
      "Q20 bases: 1572751719(99.9065%)\n",
      "Q30 bases: 1551789377(98.5749%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 38808960\n",
      "reads failed due to low quality: 46682\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 8875800\n",
      "reads with adapter trimmed: 6449900\n",
      "bases trimmed due to adapters: 128920021\n",
      "\n",
      "Duplication rate: 17.7021%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 101\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950185_1.fastq -I Data/ERR950185_2.fastq -o trimmed_fastq/ERR950185_1_trimmed_fastq.gz -O trimmed_fastq/ERR950185_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 54 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950185\n",
      "Checking: Data/ERR950186_1.fastq and Data/ERR950186_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 27746271\n",
      "total bases: 2802373371\n",
      "Q20 bases: 2560099890(91.3547%)\n",
      "Q30 bases: 2279462928(81.3404%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 27746271\n",
      "total bases: 2802373371\n",
      "Q20 bases: 2551204755(91.0373%)\n",
      "Q30 bases: 2327482393(83.054%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 17218529\n",
      "total bases: 1333657967\n",
      "Q20 bases: 1332631753(99.9231%)\n",
      "Q30 bases: 1308762605(98.1333%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 17218529\n",
      "total bases: 1360785871\n",
      "Q20 bases: 1359363668(99.8955%)\n",
      "Q30 bases: 1337664046(98.3008%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 34437058\n",
      "reads failed due to low quality: 121082\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 20934402\n",
      "reads with adapter trimmed: 8541264\n",
      "bases trimmed due to adapters: 273963332\n",
      "\n",
      "Duplication rate: 24.145%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 1\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950186_1.fastq -I Data/ERR950186_2.fastq -o trimmed_fastq/ERR950186_1_trimmed_fastq.gz -O trimmed_fastq/ERR950186_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 57 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950186\n",
      "Checking: Data/ERR950187_1.fastq and Data/ERR950187_2.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 25368011\n",
      "total bases: 2562169111\n",
      "Q20 bases: 2384491410(93.0653%)\n",
      "Q30 bases: 2156338195(84.1607%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 25368011\n",
      "total bases: 2562169111\n",
      "Q20 bases: 2370739198(92.5286%)\n",
      "Q30 bases: 2198227216(85.7956%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 19729754\n",
      "total bases: 1535952891\n",
      "Q20 bases: 1534890689(99.9308%)\n",
      "Q30 bases: 1508901083(98.2388%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 19729754\n",
      "total bases: 1603006056\n",
      "Q20 bases: 1601524474(99.9076%)\n",
      "Q30 bases: 1579316882(98.5222%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 39459508\n",
      "reads failed due to low quality: 45774\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 11230740\n",
      "reads with adapter trimmed: 4463866\n",
      "bases trimmed due to adapters: 87357795\n",
      "\n",
      "Duplication rate: 36.8104%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 1\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950187_1.fastq -I Data/ERR950187_2.fastq -o trimmed_fastq/ERR950187_1_trimmed_fastq.gz -O trimmed_fastq/ERR950187_2_trimmed_fastq.gz --qualified_quality_phred 30 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 30 --cut_front \n",
      "fastp v0.24.0, time used: 55 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950187\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Output\n",
    "mkdir -p trimmed_fastq\n",
    "\n",
    "\n",
    "# Loop through all forward reads (_R1.fastq.gz.subset.fastq) and find their corresponding reverse reads (_R2.fastq.gz.subset.fastq)\n",
    "for R1 in Data/*_1.fastq; do\n",
    "    # Extract the sample name by removing _R1.fastq.gz.subset.fastq\n",
    "    SAMPLE=$(basename \"$R1\" _1.fastq)\n",
    "    # Define file names\n",
    "    R2=\"Data/${SAMPLE}_2.fastq\"         # Reverse read file\n",
    "\n",
    "    echo \"Checking: $R1 and $R2\"\n",
    "\n",
    "    OUT_R1=\"trimmed_fastq/${SAMPLE}_1_trimmed_fastq.gz\"\n",
    "    OUT_R2=\"trimmed_fastq/${SAMPLE}_2_trimmed_fastq.gz\"\n",
    "    # Run fastp, input for -i and -I (Forward and Reverse reads), output into OUT_ R1 and R2. Filter scores under 30, and better trimming with cut_right\n",
    "    fastp -i \"$R1\" -I \"$R2\" -o \"$OUT_R1\" -O \"$OUT_R2\" \\\n",
    "        --qualified_quality_phred 30 \\\n",
    "        --cut_right \\\n",
    "        --cut_right_window_size 4 \\\n",
    "        --cut_right_mean_quality 30 \\\n",
    "        --cut_front \n",
    "\n",
    "    echo \"Finished processing: $SAMPLE\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88718668-588b-4f04-ab76-2a18a6b21a3d",
   "metadata": {},
   "source": [
    "# FastQC and multiQC on processed reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1707942e-32cd-4bab-8e61-7463cf72a9fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of ERR950168_1_trimmed_fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Approx 5% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950168_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950168_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950168_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950168_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950168_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950169_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950169_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950169_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950169_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950169_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950169_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950170_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950170_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950170_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950170_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950170_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950170_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950177_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950177_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950177_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950177_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950177_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950177_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950178_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950178_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950178_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950178_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950178_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950178_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950179_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950179_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950179_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950179_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950179_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950179_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950185_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950185_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950185_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950185_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950185_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950185_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950186_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950186_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950186_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950186_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950186_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950186_2_trimmed_fastq.gz\n",
      "Started analysis of ERR950187_1_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950187_1_trimmed_fastq.gz\n",
      "Analysis complete for ERR950187_1_trimmed_fastq.gz\n",
      "Started analysis of ERR950187_2_trimmed_fastq.gz\n",
      "Approx 5% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 10% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 15% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 20% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 25% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 30% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 35% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 40% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 45% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 50% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 55% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 60% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 65% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 70% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 75% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 80% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 85% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 90% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Approx 95% complete for ERR950187_2_trimmed_fastq.gz\n",
      "Analysis complete for ERR950187_2_trimmed_fastq.gz\n",
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=916976;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ ðŸ” \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/FastQC_aftertrim\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m36/36\u001b[0m  l\u001b[0mm  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 18 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : aftertrim_multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : aftertrim_multiqc.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "#Fastqc of the fastp dataset\n",
    "! mkdir -p FastQC_aftertrim\n",
    "! fastqc trimmed_fastq/*_fastq.gz -o FastQC_aftertrim/\n",
    "! multiqc FastQC_aftertrim/ -n aftertrim_multiqc.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "984834a4-e982-4683-9c3c-48eab0a21265",
   "metadata": {},
   "source": [
    "# Building STAR reference geneome\n",
    "Genome from https://www.ensembl.org/Homo_sapiens/Info/Index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "647a72a7-7ca1-4350-ab48-32f2d9bc2e79",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "gunzip Reference/*.gz \n",
    "\n",
    "mkdir -p human_index\n",
    "\n",
    "STAR --runMode genomeGenerate --genomeDir human_index/ --genomeFastaFiles Reference/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa \\\n",
    "    --sjdbGTFfile Reference/Homo_sapiens.GRCh38.113.gtf --runThreadN 8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f02e264a-e6e7-419a-8b30-2688898ed1a6",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7616ba61-31c6-4eda-8376-59f4d33086e2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a92ce80c-28ab-43d0-b662-743bf9357911",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'PatientNumber' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mNameError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[18]\u001b[39m\u001b[32m, line 4\u001b[39m\n\u001b[32m      2\u001b[39m \u001b[38;5;66;03m#df = pd.read_csv(\"gene_counts.txt\", sep=\"\\t\", comment=\"#\")\u001b[39;00m\n\u001b[32m      3\u001b[39m patient_list = pd.read_csv(\u001b[33m\"\u001b[39m\u001b[33mpatient_list.csv\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m----> \u001b[39m\u001b[32m4\u001b[39m \u001b[38;5;28mprint\u001b[39m(patient_list.loc[patient_list[\u001b[43mPatientNumber\u001b[49m==\u001b[33m\"\u001b[39m\u001b[33mMet\u001b[39m\u001b[33m\"\u001b[39m]])\n",
      "\u001b[31mNameError\u001b[39m: name 'PatientNumber' is not defined"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "#df = pd.read_csv(\"gene_counts.txt\", sep=\"\\t\", comment=\"#\")\n",
    "patient_list = pd.read_csv(\"patient_list.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1bf86dec-278b-43b0-81b0-77820b777d35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 ERR950168  ERR950169  ERR950170  ERR950177  ERR950178  \\\n",
      "Geneid                                                                   \n",
      "ENSG00000142611        600        640        447        706        984   \n",
      "ENSG00000284616          0          0          0          0          0   \n",
      "ENSG00000157911         45        113         45         56         25   \n",
      "ENSG00000260972          0          0          0          0          0   \n",
      "ENSG00000224340          0          0          0          3          4   \n",
      "\n",
      "                 ERR950179  ERR950185  ERR950186  ERR950187  \n",
      "Geneid                                                       \n",
      "ENSG00000142611        874        345        765        704  \n",
      "ENSG00000284616          0          0          0          0  \n",
      "ENSG00000157911         27         57         32         57  \n",
      "ENSG00000260972          0          0          0          0  \n",
      "ENSG00000224340          1          0          2          2  \n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load featureCounts file, skipping header lines (lines starting with \"#\")\n",
    "gene_counts_file = \"gene_counts.txt\"\n",
    "df = pd.read_csv(gene_counts_file, sep=\"\\t\", comment=\"#\")\n",
    "\n",
    "# Rename the columns by extracting patient names from file paths\n",
    "df.columns = df.columns.str.replace(r'.*/JOB_TMP/(\\w+)_Aligned.sortedByCoord.out.bam', r'\\1', regex=True)\n",
    "\n",
    "# Set Geneid as index\n",
    "df.set_index(\"Geneid\", inplace=True)\n",
    "\n",
    "\n",
    "df.to_csv(\"gene_counts_cleaned.csv\")\n",
    "\n",
    "counts = df.drop(columns=['Chr','Start','End','Length','Strand'])\n",
    "\n",
    "# Save as CSV for later use\n",
    "counts.to_csv(\"counts_data.csv\")\n",
    "\n",
    "# Display the first few row\n",
    "print(counts.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86197db1-9f0e-4675-9af2-8d9a01e90880",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
