{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b492339c-7a3c-4e55-a515-f9a86e20a44c",
   "metadata": {},
   "source": [
    "# Gastric Warriors Differential gene expression analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c611cf59-304e-4336-900c-819ac8bdac4d",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "Importing correct files: Using bash\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "fa6fbc16-09cb-41b2-baef-2a8cddee4204",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working directory set to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/Data/\n",
      "RNAseq-data in /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "workdir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/Data/\"\n",
    "echo \"Working directory set to $workdir\"\n",
    "\n",
    "RNAlist=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt\"\n",
    "RNAdata=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/\"\n",
    "echo \"RNAseq-data in ${RNAdata}\"\n",
    "\n",
    "\n",
    "grep \"Met\" $RNAlist | cut -f 1 > Metaplasia_patients.txt\n",
    "grep \"Hp\" $RNAlist | cut -f 1 >> Metaplasia_patients.txt\n",
    "\n",
    "#find $RNAdata -type f -name \"*subset.fastq*\" | grep -Ff Metaplasia_patients.txt | xargs -I {} cp {} \"$workdir\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "467126d3-f738-4a95-8e3f-3f47ce105660",
   "metadata": {},
   "source": [
    "Create a CSV with patient list and health condition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c6e7258f-ad5c-4336-9754-299be544c7d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Create patient list\n",
    "RNAlist=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt\"\n",
    "\n",
    "# Define the output file\n",
    "OUTPUT=\"patient_list.csv\"\n",
    "# Add CSV header\n",
    "echo \"PatientNumber,HealthCondition\" > $OUTPUT\n",
    "\n",
    "# Create patient_list.txt with \"Met\" and \"Hp\" conditions\n",
    "awk '{print $1, \",Met\"}' <(grep \"Met\" $RNAlist | cut -f1) >> $OUTPUT\n",
    "awk '{print $1, \",Hp\"}' <(grep \"Hp\" $RNAlist | cut -f1) >> $OUTPUT\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccf9c36f-dbff-414f-a8c6-d2218d7ed4ac",
   "metadata": {},
   "source": [
    "Now decompress all the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a5acecd9-5780-4b4f-85b8-2326f9eedf39",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#set the working directory\n",
    "workdir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/Data/\"\n",
    "\n",
    "#use gzip to decrompress the files\n",
    "gzip -d \"${workdir}\"*\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0c8c86e-9873-410c-b93e-97e4d3f4ebd2",
   "metadata": {},
   "source": [
    "# MultiQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0b15720e-c086-4482-8aaf-1232d5f21787",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "null\n",
      "null\n",
      "Started analysis of ERR950168_1.fastq.gz.subset.fastq\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "Approx 5% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "Approx 10% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950168_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950168_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950169_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950169_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950170_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950170_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950177_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950177_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950178_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950178_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950179_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950179_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950185_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950185_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950186_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950186_2.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950187_1.fastq.gz.subset.fastq\n",
      "Started analysis of ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 5% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 10% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 15% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 20% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 25% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 30% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 35% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 40% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 45% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 50% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 55% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 60% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 65% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 70% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 75% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 80% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 85% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 90% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 95% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Approx 100% complete for ERR950187_2.fastq.gz.subset.fastq\n",
      "Analysis complete for ERR950187_2.fastq.gz.subset.fastq\n"
     ]
    }
   ],
   "source": [
    "#First generate fastQC reports, all files in Data and output to /subset_fastqc/\n",
    "! fastqc Data/*.subset.fastq -o subset_fastqc/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7230731f-8a91-4d92-83f3-9c8c23a2753d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=395354;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/subset_fastqc\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m36/36\u001b[0m  p\u001b[0mm  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 18 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : multiqc_pre_subset_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : multiqc_pre_subset.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "#Take entire subset_fastqc/ and generate multiqc-file\n",
    "! multiqc subset_fastqc/ -n multiqc_pre_subset.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfc7d074-0afe-4621-889d-7d4866590f0e",
   "metadata": {},
   "source": [
    "# Preprocessing using FastP\n",
    "> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b1e71cd2-4aaa-4712-ba8b-f17a4f039dca",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking: Data/ERR950168_1.fastq.gz.subset.fastq and Data/ERR950168_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 49033765(97.0966%)\n",
      "Q30 bases: 46201084(91.4873%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48644322(96.3254%)\n",
      "Q30 bases: 46586827(92.2511%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 471900\n",
      "total bases: 45504357\n",
      "Q20 bases: 44769741(98.3856%)\n",
      "Q30 bases: 42633528(93.6911%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 471900\n",
      "total bases: 45504357\n",
      "Q20 bases: 44928366(98.7342%)\n",
      "Q30 bases: 43452519(95.4909%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 943800\n",
      "reads failed due to low quality: 53744\n",
      "reads failed due to too many N: 2456\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 201558\n",
      "bases trimmed due to adapters: 4412742\n",
      "\n",
      "Duplication rate: 15.1342%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950168_1.fastq.gz.subset.fastq -I Data/ERR950168_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950168_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950168_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950168\n",
      "Checking: Data/ERR950169_1.fastq.gz.subset.fastq and Data/ERR950169_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 45991156(91.0716%)\n",
      "Q30 bases: 40593825(80.3838%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48440432(95.9216%)\n",
      "Q30 bases: 46774237(92.6223%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 455588\n",
      "total bases: 43605416\n",
      "Q20 bases: 40825536(93.6249%)\n",
      "Q30 bases: 36693088(84.148%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 455588\n",
      "total bases: 43605416\n",
      "Q20 bases: 43225131(99.1279%)\n",
      "Q30 bases: 42166906(96.7011%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 911176\n",
      "reads failed due to low quality: 87002\n",
      "reads failed due to too many N: 1822\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 238964\n",
      "bases trimmed due to adapters: 4961722\n",
      "\n",
      "Duplication rate: 10.1618%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950169_1.fastq.gz.subset.fastq -I Data/ERR950169_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950169_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950169_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950169\n",
      "Checking: Data/ERR950170_1.fastq.gz.subset.fastq and Data/ERR950170_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 46218872(91.5225%)\n",
      "Q30 bases: 40966232(81.1213%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48670970(96.3782%)\n",
      "Q30 bases: 47036701(93.142%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 459072\n",
      "total bases: 44471881\n",
      "Q20 bases: 41692458(93.7502%)\n",
      "Q30 bases: 37499404(84.3216%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 459072\n",
      "total bases: 44471881\n",
      "Q20 bases: 44072216(99.1013%)\n",
      "Q30 bases: 42984745(96.656%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 918144\n",
      "reads failed due to low quality: 79968\n",
      "reads failed due to too many N: 1888\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 174702\n",
      "bases trimmed due to adapters: 3883654\n",
      "\n",
      "Duplication rate: 10.865%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950170_1.fastq.gz.subset.fastq -I Data/ERR950170_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950170_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950170_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950170\n",
      "Checking: Data/ERR950177_1.fastq.gz.subset.fastq and Data/ERR950177_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48400761(95.8431%)\n",
      "Q30 bases: 45191519(89.4882%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 47452953(93.9662%)\n",
      "Q30 bases: 45040013(89.1881%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 456890\n",
      "total bases: 44608238\n",
      "Q20 bases: 43577624(97.6896%)\n",
      "Q30 bases: 41171367(92.2954%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 456890\n",
      "total bases: 44608238\n",
      "Q20 bases: 43717663(98.0036%)\n",
      "Q30 bases: 42007504(94.1698%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 913780\n",
      "reads failed due to low quality: 84788\n",
      "reads failed due to too many N: 1432\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 142440\n",
      "bases trimmed due to adapters: 3177484\n",
      "\n",
      "Duplication rate: 15.2336%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950177_1.fastq.gz.subset.fastq -I Data/ERR950177_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950177_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950177_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950177\n",
      "Checking: Data/ERR950178_1.fastq.gz.subset.fastq and Data/ERR950178_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48575767(96.1896%)\n",
      "Q30 bases: 45376002(89.8535%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48173231(95.3925%)\n",
      "Q30 bases: 45991648(91.0726%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 465374\n",
      "total bases: 43493522\n",
      "Q20 bases: 42638014(98.033%)\n",
      "Q30 bases: 40457755(93.0202%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 465374\n",
      "total bases: 43493522\n",
      "Q20 bases: 42880474(98.5905%)\n",
      "Q30 bases: 41536346(95.5001%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 930748\n",
      "reads failed due to low quality: 67696\n",
      "reads failed due to too many N: 1556\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 283034\n",
      "bases trimmed due to adapters: 7220542\n",
      "\n",
      "Duplication rate: 6.5548%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 103\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950178_1.fastq.gz.subset.fastq -I Data/ERR950178_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950178_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950178_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950178\n",
      "Checking: Data/ERR950179_1.fastq.gz.subset.fastq and Data/ERR950179_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48926589(96.8843%)\n",
      "Q30 bases: 46148778(91.3837%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48449206(95.939%)\n",
      "Q30 bases: 46521495(92.1218%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 467275\n",
      "total bases: 45355714\n",
      "Q20 bases: 44605628(98.3462%)\n",
      "Q30 bases: 42488516(93.6784%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 467275\n",
      "total bases: 45355714\n",
      "Q20 bases: 44814472(98.8067%)\n",
      "Q30 bases: 43440444(95.7772%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 934550\n",
      "reads failed due to low quality: 63000\n",
      "reads failed due to too many N: 2450\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 179086\n",
      "bases trimmed due to adapters: 3780416\n",
      "\n",
      "Duplication rate: 8.5562%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950179_1.fastq.gz.subset.fastq -I Data/ERR950179_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950179_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950179_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950179\n",
      "Checking: Data/ERR950185_1.fastq.gz.subset.fastq and Data/ERR950185_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48495631(96.031%)\n",
      "Q30 bases: 45255730(89.6153%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 47966498(94.9832%)\n",
      "Q30 bases: 45777472(90.6485%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 462683\n",
      "total bases: 44279348\n",
      "Q20 bases: 43313656(97.8191%)\n",
      "Q30 bases: 40984532(92.559%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 462683\n",
      "total bases: 44279348\n",
      "Q20 bases: 43592903(98.4497%)\n",
      "Q30 bases: 42146487(95.1832%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 925366\n",
      "reads failed due to low quality: 72994\n",
      "reads failed due to too many N: 1640\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 209736\n",
      "bases trimmed due to adapters: 5046526\n",
      "\n",
      "Duplication rate: 7.2142%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950185_1.fastq.gz.subset.fastq -I Data/ERR950185_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950185_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950185_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 3 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950185\n",
      "Checking: Data/ERR950186_1.fastq.gz.subset.fastq and Data/ERR950186_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 46197465(91.4801%)\n",
      "Q30 bases: 40838552(80.8684%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48655120(96.3468%)\n",
      "Q30 bases: 46913618(92.8983%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 459045\n",
      "total bases: 44515737\n",
      "Q20 bases: 41690952(93.6544%)\n",
      "Q30 bases: 37398979(84.0129%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 459045\n",
      "total bases: 44515737\n",
      "Q20 bases: 44107438(99.0828%)\n",
      "Q30 bases: 42937108(96.4538%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 918090\n",
      "reads failed due to low quality: 80044\n",
      "reads failed due to too many N: 1866\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 174986\n",
      "bases trimmed due to adapters: 3780002\n",
      "\n",
      "Duplication rate: 13.6372%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950186_1.fastq.gz.subset.fastq -I Data/ERR950186_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950186_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950186_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 2 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950186\n",
      "Checking: Data/ERR950187_1.fastq.gz.subset.fastq and Data/ERR950187_2.fastq.gz.subset.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48869499(96.7713%)\n",
      "Q30 bases: 45832198(90.7568%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 500000\n",
      "total bases: 50500000\n",
      "Q20 bases: 48336303(95.7155%)\n",
      "Q30 bases: 46116041(91.3189%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 467050\n",
      "total bases: 45717950\n",
      "Q20 bases: 44879626(98.1663%)\n",
      "Q30 bases: 42512897(92.9895%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 467050\n",
      "total bases: 45717950\n",
      "Q20 bases: 45024034(98.4822%)\n",
      "Q30 bases: 43380186(94.8866%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 934100\n",
      "reads failed due to low quality: 63604\n",
      "reads failed due to too many N: 2296\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 149058\n",
      "bases trimmed due to adapters: 2985616\n",
      "\n",
      "Duplication rate: 20.043%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 124\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i Data/ERR950187_1.fastq.gz.subset.fastq -I Data/ERR950187_2.fastq.gz.subset.fastq -o trimmed_fastq_subset/ERR950187_1_trimmed_subset.fastq.gz -O trimmed_fastq_subset/ERR950187_2_trimmed_subset.fastq.gz --qualified_quality_phred 30 \n",
      "fastp v0.24.0, time used: 1 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished processing: ERR950187\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Output\n",
    "mkdir -p trimmed_fastq_subset\n",
    "\n",
    "\n",
    "# Loop through all forward reads (_R1.fastq.gz.subset.fastq) and find their corresponding reverse reads (_R2.fastq.gz.subset.fastq)\n",
    "for R1 in Data/*_1.fastq.gz.subset.fastq; do\n",
    "    # Extract the sample name by removing _R1.fastq.gz.subset.fastq\n",
    "    SAMPLE=$(basename \"$R1\" _1.fastq.gz.subset.fastq)\n",
    "    # Define file names\n",
    "    R2=\"Data/${SAMPLE}_2.fastq.gz.subset.fastq\"         # Reverse read file\n",
    "\n",
    "    echo \"Checking: $R1 and $R2\"\n",
    "\n",
    "    OUT_R1=\"trimmed_fastq_subset/${SAMPLE}_1_trimmed_subset.fastq.gz\"\n",
    "    OUT_R2=\"trimmed_fastq_subset/${SAMPLE}_2_trimmed_subset.fastq.gz\"\n",
    "    # Run fastp\n",
    "    fastp -i \"$R1\" -I \"$R2\" -o \"$OUT_R1\" -O \"$OUT_R2\" --qualified_quality_phred 30\n",
    "\n",
    "    echo \"Finished processing: $SAMPLE\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1707942e-32cd-4bab-8e61-7463cf72a9fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of ERR950168_1_trimmed_subset.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Approx 5% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950168_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950168_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950169_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950169_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950170_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950170_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950177_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950177_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950178_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950178_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950179_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950179_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950185_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950185_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950186_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950186_2_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950187_1_trimmed_subset.fastq.gz\n",
      "Started analysis of ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 5% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 10% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 15% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 20% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 25% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 30% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 35% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 40% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 45% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 50% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 55% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 60% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 65% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 70% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 75% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 80% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 85% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 90% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Approx 95% complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "Analysis complete for ERR950187_2_trimmed_subset.fastq.gz\n",
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=325917;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_gaswar/FastQC_aftertrim_subset\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m36/36\u001b[0m  p\u001b[0mm  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 18 reports\n",
      "\u001b[34m     write_results\u001b[0m | Existing reports found, adding suffix to filenames. Use '--force' to overwrite.\n",
      "\u001b[34m     write_results\u001b[0m | Data        : aftertrim_multiqc_subset_data_1\n",
      "\u001b[34m     write_results\u001b[0m | Report      : aftertrim_multiqc_subset_1.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "#Fastqc of the fastp dataset\n",
    "! mkdir -p FastQC_aftertrim_subset\n",
    "! fastqc trimmed_fastq_subset/*_subset.fastq.gz -o FastQC_aftertrim_subset/\n",
    "! multiqc FastQC_aftertrim_subset/ -n aftertrim_multiqc_subset.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6bbd9dad-af49-4496-9210-53964ebd8499",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a92ce80c-28ab-43d0-b662-743bf9357911",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'pandas'",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mModuleNotFoundError\u001b[39m                       Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[6]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[38;5;28;01mimport\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01mpandas\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mas\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01mpd\u001b[39;00m\n\u001b[32m      3\u001b[39m patient_list = pd.read_csv(patient_list.csv)\n\u001b[32m      4\u001b[39m \u001b[38;5;28mprint\u001b[39m(patient_list)\n",
      "\u001b[31mModuleNotFoundError\u001b[39m: No module named 'pandas'"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "patient_list = pd.read_csv(patient_list.csv)\n",
    "print(patient_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1bf86dec-278b-43b0-81b0-77820b777d35",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
